distribution/lib/Standard/Base/0.0.0-dev/src/Data/Statistics.enso

```diff
@@ -279,3 +319,46 @@ Vector.compute self statistic=Statistic.Count =
 Vector.compute_bulk : Vector Statistic -> Vector Any
 Vector.compute_bulk self statistics=[Statistic.Count, Statistic.Sum] =
     Statistic.compute_bulk self statistics
+
+## Compute a single running statistic on the vector.
+
+   Arguments:
+   - statistic: Statistic to calculate.
+Vector.running : Statistic -> Vector Any
+Vector.running self statistic=Statistic.Count =
+    Statistic.running self statistic
+
+## PRIVATE
+compute_fold c v =
+    if is_valid v . not then c else
+         c.first.increment
+
+         if c.first.comparatorError.not then
+             if c.first.minimum.is_nothing then c.first.setMinimum v else
+                 ordering = Incomparable_Values.handle_errors <| v.compare_to c.first.minimum
+                 if ordering.is_error then c.first.failComparator else
+                     if ordering == Ordering.Less then c.first.setMinimum v
+             if c.first.maximum.is_nothing then c.first.setMaximum v else
+                 ordering = Incomparable_Values.handle_errors <| v.compare_to c.first.maximum
+                 if ordering.is_error then c.first.failComparator else
+                     if ordering == Ordering.Greater then c.first.setMaximum v
+
+         new_moments = if c.second.is_nothing then Nothing else case v of
+             _ : Number -> c.second.add v
+             _ -> Nothing
```

Comment:
This looks like a lot of null checks.

I'm wondering if it really is faster to compute both counter and moments in one iteration? If we are ignoring moments we are doing a lot of branching here.

Wouldn't it be better to have two iterations, one for counter and one, optional, for the moments? Well, I guess the only way to find out which is faster would be to benchmark it though, as intuition can probably be unreliable. So I guess I don't want to suggest modifications here, but just thinking that this piece of code is a bit complicated as it is doing so many things at once.